### PR TITLE
CC-16236 configurable bundle

### PR DESCRIPTION
**Pull request by pyvovarov-s**
CC-16236 configurable bundle

### CODE REVIEW

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+# All suite fails fue to the BUG https://spryker.atlassian.net/browse/CC-16691
+####### GET #######
+Get_configurable_bundle_templates_by_invalid_configurable_bundle_template_id
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:13]: this test does not seem to fail on staging - is the bug still relevant?
also why is this test here? why do we have configured-bundles and configurable-bundle-templates endpoints in one file? can we divide them to different filed in this folder?

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    ...  AND    Find or create customer cart
+    ...  AND    Cleanup all items in the cart:    ${cart_id}
+    When I send a POST request:    /carts/${cart_id}/configured-bundles    {"data": {"type": "configured-bundles","attributes": {"quantity": "","templateUuid": "","items": [{"sku": "","quantity": 2,"slotUuid": ""}]}}}
+    Then Response status code should be:    422
+    And Response reason should be:    Unprocessable Content
+    And Response should return error code:    4003
+    And Response should return error message:    The quantity of the configured bundle should be more than zero.
+
+####### PATCH #######
+Update_configured_bundle_item_in_cart_with_non_existing_bundle_group_key
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:98]: update tests with invalid body (e.g. non-integer qty) are missing

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+Test Setup        TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+# All suite fails fue to the BUG https://spryker.atlassian.net/browse/CC-16691
+Get_configurable_bundle_templates
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot:12]: this test does not fail on staging - is this bug still relevant?

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot (hunk 1 above)
Comment (himanshu-saxena-metacube) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot]: Please check again
1. Test does not fails on staging, hence removed bud id
2. Removed configurable-bundle-templates endpoints

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot (hunk 2 above)
Comment (himanshu-saxena-metacube) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:101]: Fixed

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot (hunk 3 above)
Comment (himanshu-saxena-metacube) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot:11]: Removed bug id as test does not fails on staging

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot (hunk 1 above)
Comment (OKrivtsova) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot]: you should not remove the configurable-bundle-templates tests. I do not know who is working on configurable-bundle-templates endpoint - if it is you - do not delete them, just move to another subfolder. If it is someone else - remove them all

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot (hunk 2 above)
Comment (OKrivtsova) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:101]: you added test for ADD with ivalid qty, but I see not test for updating with invalid qty - please add it

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot (hunk 1 above)
Comment (himanshu-saxena-metacube) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot]: We have removed configurable-bundle-templates tests for now as Gorav is already committed his changes seperatly.

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot (hunk 2 above)
Comment (himanshu-saxena-metacube) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:101]: Fixed

File: tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    And Response should return error code:    4004
+    And Response should return error message:    Configured bundle with provided group key not found in cart.
+
+Update_configured_bundle_item_in_cart_with_invalid_qty
+  [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+      ...  AND    I set Headers:    Authorization=${token}
+      ...  AND    Find or create customer cart
+      ...  AND    Cleanup all items in the cart:    ${cart_id}
+    ...    AND     I send a POST request:    /carts/${cart_id}/configured-bundles?include=items    {"data": {"type": "configured-bundles","attributes": {"quantity": "${configured_bundle_quantity}","templateUuid": "${configurable_bundle_template_id}","items": [{"sku": "${configurable_bundle_first_slot_item_sku}","quantity": 2,"slotUuid": "${configurable_bundle_first_slot_uuid}"}]}}}
+    When I send a PATCH request:    /carts/${cart_id}/configured-bundles     {"data": {"type": "configured-bundles","attributes": {"quantity": "abcd"}}}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:118]: this is not a valid test - patch is missing config bundle id, so the error is 'Resource id is not specified.'
the error should be 'The quantity of the configured bundle should be more than zero.' please fix it

Comment (himanshu-saxena-metacube) [reply tests/api/b2c/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot]: Fixed, Pl check

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**